darwin/dataset/utils.py

```diff
@@ -0,0 +1,291 @@
+import json
+import os
+from collections import defaultdict
+from pathlib import Path
+from typing import List, Optional, Iterable
+from sklearn.model_selection import train_test_split
```

Comment:
We are only depending on `sklearn` to use `train_test_split`. Is it worth it?

Reply:
Valid point. [Here](https://github.com/scikit-learn/scikit-learn/blob/1495f6924/sklearn/model_selection/_split.py#L1998) you can see the function. I think extracting that would be non-trivial (for me at least). Is there a way to do it nicely?

Reply:
If the dependency is not too big, I guess we can always change it later, what do you think @simedw